### DISCUSSION TITLE
More Skin Editor Change Requests

### DISCUSSION THREAD

**Discussion by TheFantasticLoki** [ppy/osu#19767] [Ideas]
Finally took another dive into the skin editor in lazer after SEVERAL months and I'm glad to see the addition of some more legacy skinables and skinable sprites but I still have a few descrepancies with the current state of the editor and have a few more ideas to add some polish and usability improvements.

First off: PROPER Layer Support, Currently when adding elements to the active scene they are placed on the top most layer which makes perfect sense and is as it should be, BUT there is absolutely no way to move an element down layers or to move an existing element up layers, etc. Why would you want to move elements to other layers? Simple, existing design choices. Many skins, mine included, have purpose designed elements to restrict the view of certain existing assets such as song progression meter shown below:
![image](https://user-images.githubusercontent.com/26889568/184581981-66840d9b-76d2-4d01-94a1-567e2f569346.png)
Another huge reason is that newly added elements can block the view of existing elements especially with the addition of skinable sprites. Since there is no way to move an element up or down layers, in order to fix this issue you must replace all blocked elements until your layering is how you want it. This can be EXTREMELY frustrating especially with fullscreen assets like many hp bars as the play area scaling is VERY hard to match up within the editor. An addition of simple layer up and down buttons would suffice in resolving this gripe as we don't technically need a dedicated layers window.

Second: We need a skin preview screen where we can preview the output of the skin with different aspect ratios etc for easier verification of alignment without having to constantly exit skin editor to check alignment. This has been a gripe of mine since the skin editor was first introduced. While the new way of automatic skin scaling is nice and works well when creating a new skin from scratch, it doesn't work well with many existing skins with design elements that include the whole screen for play area backgrounds and other various elements like the key overlay. It's especially frustrating since we cannot hide the components or element settings bars on either side of the screen to free up space for resizing since we have to expand the size of the elements past the bounds of the play area so they connect properly with the sides of the screen and fit as they did in stable.
While you may simply say just use a different skin or fix it yourself, for many players we have no proper experience with photo manipulation and many players have spent actual money to have skins hand crafted for them exclusively and losing these investments is not a route in which I want to see the game head towards.

Third: Still missing some legacy components such as legacy key overlay. While this isn't a major gripe it's something I have genuinely missed as my skin has a custom design for the stable key overlay and on lazer that space is just empty.

**Comment by peppy** [comment]
Mostly duplicate
